realm/realm-library/src/androidTest/java/io/realm/RxJavaTests.java

```diff
@@ -1058,29 +844,264 @@ public void dynamicRealmObject_gcStressTest() {
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void asChangesetObservable_frozenRealmObject() {
+        realm.beginTransaction();
+        final AllJavaTypes obj = realm.createObject(AllJavaTypes.class, 42);
+        realm.commitTransaction();
+
+        subscription = obj.<AllJavaTypes>freeze().<AllJavaTypes>asChangesetObservable().subscribe(change -> {
+            AllJavaTypes rxObject = change.getObject();
+            assertTrue(rxObject.isFrozen());
+            assertNotEquals(rxObject, obj);
+            assertEquals(rxObject.getFieldId(), obj.getFieldId());
+            looperThread.testComplete();
+        });
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void asFlowable_frozenDynamicRealmObject() {
+        realm.beginTransaction();
+        realm.createObject(AllTypes.class);
+        realm.commitTransaction();
+
+        DynamicRealm dynRealm = DynamicRealm.getInstance(looperThread.getConfiguration());
+        DynamicRealmObject obj = dynRealm.where(AllTypes.CLASS_NAME).findFirst();
+
+        DynamicRealmObject frozenObject = obj.<DynamicRealmObject>freeze();
+        subscription = frozenObject.asFlowable().subscribe(rxObject -> {
+            assertTrue(rxObject.isFrozen());
+            assertEquals(rxObject, frozenObject);
+            looperThread.testComplete();
+        });
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void asChangesetObservable_frozenDynamicRealmObject() {
+        realm.beginTransaction();
+        realm.createObject(AllTypes.class);
+        realm.commitTransaction();
+
+        DynamicRealm dynRealm = DynamicRealm.getInstance(looperThread.getConfiguration());
+        DynamicRealmObject obj = dynRealm.where(AllTypes.CLASS_NAME).findFirst();
+
+        DynamicRealmObject frozenObject = obj.<DynamicRealmObject>freeze();
+        subscription = frozenObject.asChangesetObservable().subscribe(change -> {
+            RealmObject rxObject = change.getObject();
+            assertTrue(rxObject.isFrozen());
+            assertEquals(rxObject, frozenObject);
+            assertNull(change.getChangeset());
+            looperThread.testComplete();
+        });
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void realmResults_readableAcrossThreads() {
+        // FIXME: Make a test like this for all from methods
+        final long TEST_SIZE = 10;
+        Realm realm = looperThread.getRealm();
+
+        realm.beginTransaction();
+        for (int i = 0; i < TEST_SIZE; i++) {
+            realm.createObject(AllTypes.class).setColumnLong(1);
+        }
+        realm.commitTransaction();
+
+        subscription = realm.where(AllTypes.class).sort(AllTypes.FIELD_LONG).findAllAsync().asFlowable()
+                .subscribeOn(Schedulers.io())
+                .filter(RealmResults::isLoaded)
+                .map(results -> {
+                    RealmLog.error(Thread.currentThread().getName());
+                    return new Pair<>(results.size(), results);
```

Comment:
can you pass also one or multiple frozen objects so we can assert that observing them from another thread is ok